### ISSUE TITLE
UserSession.getUser() does not use the provided fetch

### ISSUE THREAD

**Issue by haoliangyu** [Esri/arcgis-rest-js#276]
Thanks for working on #203. I try the new version (v1.7.0) and find a small gap in the code.

The `UserSession.getUser()` does not accept the provided fetch implementation (or any request option?). This affects the item sharing function in `@esri/arcgis-rest-sharing`:

``` javascript
const arcgisRestSharing = require('@esri/arcgis-rest-sharing');
const fetch = require('node-fetch');

// some code to create a userSession

const params = {
    authentication: userSession,
    id: "my item",
    groupId: "my group",
    fetch
}

arcgisRestSharing.shareItemWithGroup(params);
```

Then it throws an error:

``` 
Error: `arcgis-rest-request` requires global variables for `fetch`, `Promise` and `FormData` to be present in the global scope. You are missing `fetch`. We recommend installing the `isomorphic-fetch` modules at the root of your application to add these to the global scope. See https://bit.ly/2KNwWaJ for more info.
    at Object.request (/my-app/node_modules/@esri/arcgis-rest-request/dist/node/request.js:69:15)
    at UserSession.getUser (/my-app/node_modules/@esri/arcgis-rest-auth/dist/node/UserSession.js:307:42)
    at Object.isOrgAdmin (/my-app/node_modules/@esri/arcgis-rest-sharing/dist/node/helpers.js:20:20)
    at changeGroupSharing (/my-app/node_modules/@esri/arcgis-rest-sharing/dist/node/group-sharing.js:55:22)
    at Object.shareItemWithGroup (/my-app/node_modules/@esri/arcgis-rest-sharing/dist/node/group-sharing.js:25:12)
    at Object.shareItemWithGroup (/my-app/packages/my-module/util.js:114:28)
    at MyApp.function (/my-app/packages/my-module/index.js:62:19)
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:228:7)
```

**Comment by jgravois** [comment]
good catch @haoliangyu.

we currently chain several intermediate requests together to determine whether or not the user is a member of the group you're attempting to share an item with or whether they are an org administrator.

the problem you identified is a result of the fact that we aren't passing through custom request options from the caller in those internal requests for metadata.

i've proposed a fix in #277

**Comment by jgravois** [comment]
released as [`v1.7.1`](https://github.com/Esri/arcgis-rest-js/releases/tag/v1.7.1) :tada:.

thx @noahmulfinger!
